### PR TITLE
Fix image optimization support for Next 14.1.1

### LINKED ISSUE #374: Next 14.1.1 Image Optimization Breaking Changes
[Open-Next Image Optimization Code](https://github.com/sst/open-next/blob/main/packages/open-next/src/adapters/image-optimization-adapter.ts#L117-L128)

[New `nextjs` code](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/image-optimizer.ts#L587-L601)

- [ ] Update `open-next` image-optimization handler to sync w/ the new breaking changes
- [ ] Add e2e test that loads an image and checks the `Content-Type` to be `image/webp`

### PR DESCRIPTION
**Pull request by chungweileong94**
fix #374

### CODE REVIEW

**Comment by conico974** [comment]
@chungweileong94 Did you have any issues with sharp version ? Like this one https://github.com/sst/open-next/issues/373

If this is not a problem anymore for 14.1.1, we could set sharp version to `0.32.2` for next 14.1.1+ and to `0.32.6` for below

**Comment by chungweileong94** [comment]
> @chungweileong94 Did you have any issues with sharp version ? Like this one https://github.com/sst/open-next/issues/373

Yes, but with different issue, and I have to downgrade to 0.32.6. I will try to verify it the issue that you mentioned.

**Comment by chungweileong94** [comment]
@conico974 Just tested `sharp@0.33.2` with `next@14.1.1`, the problem still exists.

**Comment by conico974** [comment]
> @conico974 Just tested `sharp@0.33.2` with `next@14.1.1`, the problem still exists.

Could you change sharp version here to 0.32.6 ? https://github.com/sst/open-next/blob/3deb2022d0bb506e4c4b66baeae248c7c2b153e5/packages/open-next/src/build.ts#L370
I'll try to merge and release tomorrow when i'll get some time to test things

**Comment by khuezy** [comment]
Thanks @chungweileong94 !! Can we add an e2e test to fetch an image and check that the response's Content-Type is `image/webp`? You can pick any (appropriate image) from unsplash, or https://twitter.com/thdxr/status/1738582905511669896/photo/1

File: .changeset/old-islands-invite.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"open-next": patch
+---
+
+Fix Image Optimization Support for Next@14.1.1
```
Comment (khuezy) [inline .changeset/old-islands-invite.md:5]: We probably don't want to apply the changeset now. We usually let the PR merge into main, run the e2e, and once that passes, we do a separate changeset PR.

Comment (conico974) [reply .changeset/old-islands-invite.md:5]: It should be fine actually, changesets get merged before release and the changebot don't release until we merge the version packages PR.

Comment (khuezy) [reply .changeset/old-islands-invite.md:5]: oh right, I forgot.

**Comment by khuezy** [comment]
Thanks so much @chungweileong94 !
